src/core/network.js

```diff
@@ -115,20 +133,17 @@ var NetworkManager = (function NetworkManagerClosure() {
         pendingRequest.expectedStatus = 200;
       }
 
-      if (args.onProgressiveData) {
-        // Some legacy browsers might throw an exception.
-        try {
-          xhr.responseType = 'moz-chunked-arraybuffer';
-        } catch(e) {}
-        if (xhr.responseType === 'moz-chunked-arraybuffer') {
-          pendingRequest.onProgressiveData = args.onProgressiveData;
-          pendingRequest.mozChunked = true;
-        } else {
-          xhr.responseType = 'arraybuffer';
-        }
+//#if (GENERIC || FIREFOX || MOZCENTRAL)
+      if (args.onProgressiveData && supportsMozChunked) {
+        xhr.responseType = 'moz-chunked-arraybuffer';
+        pendingRequest.onProgressiveData = args.onProgressiveData;
+        pendingRequest.mozChunked = true;
       } else {
         xhr.responseType = 'arraybuffer';
       }
+//#else
```

Comment:
We don't want to miss any other future targets (in the past it was B2G), so I'm looking into something like:

```
//#if CHROMIUM
//      xhr.responseType = 'arraybuffer';
//#endif
//#if (FIREFOX || MOZCENTRAL)
// ....
//#endif
//#if !(CHROMIUM || FIREFOX || MOZCENTRAL)
   ...
//#endif
```